File: application/src/main/java/de/pbauerochse/worklogviewer/youtrack/v20174/types/Grouping.java
package de.pbauerochse.worklogviewer.youtrack.v20174.types;

import com.fasterxml.jackson.annotation.JsonSubTypes;
import com.fasterxml.jackson.annotation.JsonTypeInfo;

@JsonTypeInfo(
        use = JsonTypeInfo.Id.NAME,
        include = JsonTypeInfo.As.PROPERTY,
        property = "$type"
)
@JsonSubTypes({
        @JsonSubTypes.Type(value = WorkItemBasedGrouping.class, name = "jetbrains.charisma.smartui.report.time.WorkItemBasedGrouping"),
        @JsonSubTypes.Type(value = FieldBasedGrouping.class, name = "jetbrains.charisma.smartui.report.time.FieldBasedGrouping")

})
@Deprecated
public interface Grouping {

    String getId();

    String getPresentation();

    GroupingField getField();

}

File: connectors/v20174_to_20181/src/main/java/de/pbauerochse/worklogviewer/connector/v2017/domain/report/Grouping.kt
package de.pbauerochse.worklogviewer.connector.v2017.domain.report

import com.fasterxml.jackson.annotation.JsonSubTypes
import com.fasterxml.jackson.annotation.JsonTypeInfo
import de.pbauerochse.worklogviewer.connector.GroupByParameter

@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "\$type", defaultImpl = UnknownGrouping::class)
@JsonSubTypes(
    JsonSubTypes.Type(value = WorkItemBasedGrouping::class, name = "jetbrains.charisma.smartui.report.time.WorkItemBasedGrouping"),
    JsonSubTypes.Type(value = FieldBasedGrouping::class, name = "jetbrains.charisma.smartui.report.time.FieldBasedGrouping")
)
interface Grouping : GroupByParameter